napari/layers/_layer_actions.py

```diff
@@ -37,7 +37,12 @@ def _duplicate_layer(ll: LayerList):
 
     for lay in list(ll.selection):
         new = deepcopy(lay)
-        new.name += ' copy'
+        # new.events.disconnect()
```

Comment:
Seems like you do this in `copy_layer` in the example, but are unsure if it's needed here? This commented line and one below should either be removed or uncommented. Or maybe left with an explanatory comment.

Reply:
yes. It is needed if I do not find the fix for deepcopy of events, as copied layer will emit signals as the source layer. 

I'm not sure if copied layer should have connected all callbacks as source. 

But when #4725 will be merged then it will be possible to make a copy of the layer without using deepcopy.